libs/tree/tree_test.go

```diff
@@ -12,6 +12,13 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+type treeEnum int
+
+const (
+	enumllrb treeEnum = iota
```

Comment:
I don't think enum here is necessary. can just declare two different tree gen functions